### PR TITLE
US 30535 - Terraform - FSLogix with ADDS

### PR DESCRIPTION
**Pull request by JamJarchitect**
# Overview/Summary
Adds the Management VM module to Terraform, including joining the storage account to the domain and a user assigned managed identity.

## This PR fixes/adds/changes/removes

1.  Adds MGMTVM module to Terraform workload.

### Breaking Changes

1. N/A

## Testing Evidence

TO BE ADDED.

## As part of this Pull Request I have

- [X] Read the [Contribution Guide](https://github.com/Azure/avdaccelerator/blob/main/CONTRIBUTING.md) and ensured this PR is compliant with the guide
- [X] Ensured the resource API versions in `.bicep` file/s I am adding/editing are using the latest API version possible
- [X] Checked for duplicate [Pull Requests](https://github.com/Azure/avdaccelerator/pulls)
- [X] Associated it with relevant [GitHub Issues](https://github.com/Azure/avdaccelerator/issues)
- [X] *(AVD LZA Team Only)* Associated it with relevant [ADO Items](https://dev.azure.com/CSUSolEng/Accelerator%20-%20AVD/_backlogs/backlog/Accelerator%20-%20AVD%20Team/Features)
- [X] Ensured my code/branch is up-to-date with the latest changes in the `main` [branch](https://github.com/Azure/avdaccelerator/tree/main)
- [X] Performed testing and provided evidence.
- [X] Updated relevant and associated documentation (e.g. Contribution Guide, Module READMEs, Docs etc.)

### CODE REVIEW

File: workload/terraform/modules/mgmtvm/main.tf [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+
+# Assigns the Managed Identity Storage Account Contributor RBAC to the Storage RG scope
+resource "azurerm_role_assignment" "assign_stguai_sac" {
+  scope                = data.azurerm_resource_group.rg_storage.id
+  role_definition_name = "Storage Account Contributor"
+  principal_id         = azurerm_user_assigned_identity.stguai.principal_id
+}
+
+# Assigns the Managed Identity Reader RBAC to the Storage RG scope
+resource "azurerm_role_assignment" "assign_stguai_sac" {
```
Comment (jensheerin) [inline workload/terraform/modules/mgmtvm/main.tf:24]: Duplicated resource name "assign_stguai_sac"

File: workload/terraform/modules/mgmtvm/main.tf [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+resource "random_string" "AVD_local_password" {
+  length           = 16
+  special          = true
+  min_special      = 2
+  override_special = "*!@#?"
+}
+
+# User Assigned Managed Identity for the MGMT VM to join storage accounts to the domain
+resource "azurerm_user_assigned_identity" "stguai" {
+  name                = "id-storage-${var.prefix}-${var.deployment_environment}-001"
```
Comment (jensheerin) [inline workload/terraform/modules/mgmtvm/main.tf:11]: input variable with the name "deployment_environment" has not been declared

File: workload/terraform/modules/mgmtvm/main.tf [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+resource "azurerm_virtual_machine_extension" "dscStorageScript" {
+  name                 = "AzureFilesDomainJoin"
+  virtual_machine_id   = azurerm_windows_virtual_machine.mgmt_vm.id
+  publisher            = "Microsoft.Compute"
+  type                 = "CustomScriptExtension"
+  type_handler_version = "1.10"
+
+  settings = jsonencode({
+    fileUris         = [var.baseScriptUri]
+    commandToExecute = "powershell.exe -ExecutionPolicy Unrestricted -File ${var.vfile} -DscPath ${var.dsc_storage_path} -StorageAccountName ${var.storage_account_name} -StorageAccountRG ${var.storage_account_rg} -StoragePurpose fslogix -DomainName ${var.domain_name} -IdentityServiceProvider ${var.IdentityServiceProvider} -AzureCloudEnvironment ${var.azure_cloud_environment} -SubscriptionId ${var.workloadSubsId} -DomainAdminUserName ${var.domain_user}@${var.domain_name} -CustomOuPath ${var.storageCustomOuPath} -OUName ${var.ouStgPath} -CreateNewOU ${var.createOuForStorageString} -ShareName ${var.file_share_name} -ClientId ${var.clientid} -DomainAdminUserPassword ${var.domain_password} -verbose"
```
Comment (jensheerin) [inline workload/terraform/modules/mgmtvm/main.tf:121]: variable with the name "storage_account_rg" has not been declare
variable with the name "workloadSubsId" has not been declared
variable with the name "storageCustomOuPath" has not been declared
variable with the name "ouStgPath" has not been declared
variable with the name "createOuForStorageString" has not been declared
variable with the name "file_share_name" has not been declared

**Review by jensheerin — CHANGES_REQUESTED** [review]
Thank you for the contributions.
See inline comments and fix. 
LMK when to review again.

File: workload/terraform/modules/mgmtvm/main.tf (hunk 1 above)
Comment (JamJarchitect) [reply workload/terraform/modules/mgmtvm/main.tf:24]: addressed - changed `sac` in code to `sar`

File: workload/terraform/modules/mgmtvm/main.tf (hunk 3 above)
Comment (JamJarchitect) [reply workload/terraform/modules/mgmtvm/main.tf:121]: addressed

File: workload/terraform/modules/mgmtvm/main.tf (hunk 2 above)
Comment (JamJarchitect) [reply workload/terraform/modules/mgmtvm/main.tf:11]: addressed

**Comment by JamJarchitect** [comment]
@jensheerin

Sorry it's taken so long. Requested changes have since been addressed - see latest push. Re-requested review

**Comment by jensheerin** [comment]
DNM
@JamJarchitect Thank you for the update. Please run validate and fix errors, then resubmit for review.

**Pull request closed without merging**